Bottom: 0f6b0b90d7739c61acae47649b0c406540570e73
Top:    41275ebf36f106cedef6963331ebce848338db3a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-17 00:56:30 +0100

buffer.c: Fix crash in SmashBufferElement

The element is allocated with g_slice_new, so must be free'd with
g_slice_free.

Broken since commit 2ce35292b9e96da38cb56878005aba20891689eb:

    Convert board objects to GLists of g_slice allocated memory


---

diff --git a/src/buffer.c b/src/buffer.c
index 8793f50..b396918 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -926,7 +926,7 @@ SmashBufferElement (BufferTypePtr Buffer)
   }
   END_LOOP;
   FreeElementMemory (element);
-  free (element);
+  g_slice_free (ElementType, element);
   return (true);
 }
